### PR TITLE
[Feature] support create or replace view

### LINKED ISSUE #27767: Support create or replace view
## Feature request

**Is your feature request related to a problem? Please describe.**

**Describe the solution you'd like**

**Describe alternatives you've considered**

**Additional context**

### PR DESCRIPTION
**Pull request by murphyatwork**
Fixes #27767

`CREATE OR REPLACE VIEW` could be used to update existed view definition. 

## What type of PR is this:
- [ ] BugFix
- [x] Feature
- [ ] Enhancement
- [ ] Refactor
- [ ] UT
- [ ] Doc
- [ ] Tool

## Checklist:
- [ ] I have added test cases for my bug fix or my new feature
- [ ] This pr will affect users' behaviors
- [ ] This pr needs user documentation (for new or modified features or behaviors)
  - [ ] I have added documentation for my new feature or new function

## Bugfix cherry-pick branch check:
- [x] I have checked the version labels which the pr will be auto-backported to the target branch
  - [x] 3.1
  - [x] 3.0
  - [x] 2.5
  - [ ] 2.4

### CODE REVIEW

**Comment by wanpengfei-git** [comment]
#### [FE PR Coverage Check]

:heart_eyes: **pass** : 28 / 33 (84.85%)




#### file detail

|   |path|covered_line|new_line|coverage|not_covered_line_detail|
|----|----|----|----|----|----|
|:large_blue_circle:|com/starrocks/server/LocalMetastore.java|21|26|80.77%|[3867, 3868, 3869, 3881, 3882]|
|:large_blue_circle:|com/starrocks/sql/ast/AlterViewStmt.java|5|5|100.00%|[]|
|:large_blue_circle:|com/starrocks/sql/ast/CreateViewStmt.java|2|2|100.00%|[]|

File: fe/fe-core/src/main/java/com/starrocks/sql/parser/StarRocks.g4 [hunk 1]
```diff
@@ -555,7 +555,7 @@ recoverPartitionStatement
 // ------------------------------------------- View Statement ----------------------------------------------------------
 
 createViewStatement
-    : CREATE VIEW (IF NOT EXISTS)? qualifiedName
+    : CREATE (OR REPLACE)? VIEW (IF NOT EXISTS)? qualifiedName
```
Comment (imay) [inline fe/fe-core/src/main/java/com/starrocks/sql/parser/StarRocks.g4:558]: Should we support "create or replace view if not exist"?
What does that mean?
I think create or replace is that create if not exist and replace if not exist.

Comment (mofeiatwork) [reply fe/fe-core/src/main/java/com/starrocks/sql/parser/StarRocks.g4:558]: fixed. it's conflicted options, we don't support it.

**Comment by wanpengfei-git** [comment]
#### [FE PR Coverage Check]

:heart_eyes: **pass** : 28 / 33 (84.85%)




#### file detail

|   |path|covered_line|new_line|coverage|not_covered_line_detail|
|----|----|----|----|----|----|
|:large_blue_circle:|com/starrocks/server/LocalMetastore.java|21|26|80.77%|[3867, 3868, 3869, 3881, 3882]|
|:large_blue_circle:|com/starrocks/sql/ast/AlterViewStmt.java|5|5|100.00%|[]|
|:large_blue_circle:|com/starrocks/sql/ast/CreateViewStmt.java|2|2|100.00%|[]|

**Review by satanson — APPROVED** [review]

**Review by kangkaisen — APPROVED** [review]

**Comment by mofeiatwork** [comment]
@Mergifyio backport branch-2.5 branch-3.0 branch-3.1

**Pull request merged**